service/controller/app/v1/resource/chartstatus/create.go

```diff
@@ -0,0 +1,49 @@
+package chartstatus
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	customResource, err := key.ToCustomResource(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	name := key.AppName(customResource)
+
+	g8sClient, err := r.kubeConfig.NewG8sClientForApp(ctx, customResource)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("getting status for chart %#q", name))
+	chart, err := g8sClient.ApplicationV1alpha1().Charts(r.watchNamespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return microerror.Maskf(notFoundError, "chart %#q in namespace %#q", name, r.watchNamespace)
+	}
+
+	if chart.Status.Status != "" && key.ReleaseStatus(customResource) != chart.Status.Status {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting chart %#q status as %#q", name, chart.Status.Status))
```

Comment:
```suggestion
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting app %#q status as %#q", name, chart.Status.Status))
```

This is setting the app status by reading the status of the chart CR. So the logging isn't accurate.

Also please add a newline to make this easier to read.